components/bl-leafletMap-component/src/index.js

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useRef } from 'react';
+
+import {
+  initMap,
+  createMarkers,
+  createCircles,
+  createPolygons,
+  changeMapType,
+  setCenter,
+  setZoom,
+  toggleZoomControl,
+  toggleDraggingControl,
+  toggleFullscreen
+} from './helpers/map';
+import { createActions } from './helpers/actions';
+import { useIcon } from './helpers/use-icon';
+import { IconOptions } from './icons';
+import { MapTypeSelect } from './map-type-select';
+import { GeopositionButton } from './geoposition-button';
+import { FullscreenButton } from './fullscreen-button';
+
+export default function LeafletMap({ component, eventHandlers }) {
+  const {
+    markers,
+    circles,
+    polygons,
+    center,
+    zoom,
+    mapType,
+    classList,
+    visibility,
+    zoomControl,
+    draggingControl,
+    mapTypeControl,
+    geopositionControl,
+    display,
+    fullscreen,
+    fullscreenControl
+  } = component;
+
+  const containerRef = useRef(null);
+  const mapRef = useRef(null);
+  const currentLayer = useRef(null);
+  const markerIcon = useIcon(IconOptions.marker);
+
+  useEffect(() => {
+    initMap(component, eventHandlers, mapRef, currentLayer);
+    createActions(component, mapRef.current);
+  }, []);
+
+  useEffect(() => {
```

Comment:
wouldn't be better to create custom hooks to control the map instead of using a plain useEffect and a util function?

```
function useMapType(component, mapRef){
  useEffect(() => {
    if (mapType !== undefined && mapRef.current) {
      if (mapType in Maps) {
       map.current.removeLayer(currentLayer.current);

       currentLayer.current = Leaflet.tileLayer(Maps[mapType].mapUrl, Maps[mapType].options).addTo(map.current);
      } else {
        console.error('Leaflet Map: not valid map type');
      }
  }, [mapType, mapRef.current]);
}

export default function LeafletMap({ component, eventHandlers }) {
  ...
  useMapType(component, mapRef)
  useMapMarkers(...)
  useMapCenter(...)

 ...
  

...
```

Reply:
It seems to me that in the index file it is more clear what happens with the approach that is used initially.

  useEffect(() => {
    changeMapType(mapRef, currentLayer, component);
  }, [mapType]);

useEffect(() => {
    setCenter(mapRef.current, center);
  }, [center]);
  
It is clearly seen that when changing the map type, we call the changeMapType function, and when changing the center, we set a new center.
But if you think this is a mistake - I'll redo it.

Reply:
ok, let's leave it as it is